pilot/pkg/security/authz/model/generator.go

```diff
@@ -180,6 +180,35 @@ func (srcNamespaceGenerator) principal(_, value string, _ bool, useAuthenticated
 	return principalAuthenticated(m, useAuthenticated), nil
 }
 
+type srcServiceAccountGenerator struct{}
+
+func (srcServiceAccountGenerator) permission(_, _ string, _ bool) (*rbacpb.Permission, error) {
+	return nil, fmt.Errorf("unimplemented")
+}
+
+func (srcServiceAccountGenerator) principal(_, value string, _ bool, useAuthenticated bool) (*rbacpb.Principal, error) {
+	regex, err := serviceAccountRegex(value)
+	if err != nil {
+		return nil, err
+	}
+	m := matcher.StringMatcherRegex(regex)
+	return principalAuthenticated(m, useAuthenticated), nil
+}
+
+func serviceAccountRegex(value string) (string, error) {
+	ns, sa, ok := strings.Cut(value, "/")
+	if !ok {
+		return "", fmt.Errorf("invalid value: %v", value)
+	}
+	// Format should follow...
+	// 'spiffe://' then a trust domain + arbitrary k/v pairs
```

Comment:
SPIFFE [has a Go library which we should probably trust/lean on to handle "Is a valid SPIFFE ID"](https://pkg.go.dev/github.com/spiffe/go-spiffe/v2/spiffeid) checks.

I suspect our check can compose with theirs and become much simpler if we can assume we're already dealing with a valid SPIFFE ID.

Reply:
I am not sure how we can lean on this?

We are translating this to Envoy, and envoy takes a regex. The only parsing of user input we are doing in Go is `<ns>/<name>` which is trivial to parse?

Reply:
> and envoy takes a regex.

Ah, missed this was being provided to envoy as config and not something we're even  testing against ourselves.

That's good, that means this is as simple as expected to change.